compiler/rustc_mir/src/transform/validate.rs

```diff
@@ -294,6 +294,26 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                     );
                 }
             }
+            StatementKind::CopyNonOverlapping(box rustc_middle::mir::CopyNonOverlapping {
+                ref src,
+                ref dst,
+                ref count,
+            }) => {
+                let op_src_ty =
+                    src.ty(&self.body.local_decls, self.tcx).builtin_deref(true).unwrap().ty;
+                let op_dst_ty =
+                    dst.ty(&self.body.local_decls, self.tcx).builtin_deref(true).unwrap().ty;
+                // since CopyNonOverlapping is parametrized by 1 type,
+                // we only need to check that they are equal and not keep an extra parameter.
+                if op_src_ty != op_dst_ty {
+                    self.fail(location, format!("bad arg ({:?} != {:?})", op_src_ty, op_dst_ty,));
+                }
+
+                let op_cnt_ty = count.ty(&self.body.local_decls, self.tcx);
+                if op_cnt_ty != self.tcx.types.usize {
+                    self.fail(location, format!("bad arg ({:?} != usize)", op_cnt_ty))
+                }
+            }
             _ => {}
```

Comment:
I know this is preexisting, but since you have to edit again before merging anyway, please make this match exhaustive, so future additions in the style of `CopyNonoverlapping` have to add a new arm here.